src/bun.js/bindings/webcore/JSWorker.cpp

```diff
@@ -177,52 +196,28 @@ template<> JSC::EncodedJSValue JSC_HOST_CALL_ATTRIBUTES JSWorkerDOMConstructor::
             }
         }
 
-        auto workerData = optionsObject->getIfPropertyExists(lexicalGlobalObject, Identifier::fromString(vm, "workerData"_s));
+        workerData = optionsObject->getIfPropertyExists(lexicalGlobalObject, Identifier::fromString(vm, "workerData"_s));
         if (!workerData) {
             workerData = optionsObject->getIfPropertyExists(lexicalGlobalObject, Identifier::fromString(vm, "data"_s));
+            if (!workerData) workerData = jsUndefined();
         }
+        RETURN_IF_EXCEPTION(throwScope, {});
 
-        if (workerData) {
-            Vector<RefPtr<MessagePort>> ports;
-            Vector<JSC::Strong<JSC::JSObject>> transferList;
-
-            if (JSValue transferListValue = optionsObject->getIfPropertyExists(lexicalGlobalObject, Identifier::fromString(vm, "transferList"_s))) {
-                if (transferListValue.isObject()) {
-                    JSC::JSObject* transferListObject = transferListValue.getObject();
-                    if (auto* transferListArray = jsDynamicCast<JSC::JSArray*>(transferListObject)) {
-                        for (unsigned i = 0; i < transferListArray->length(); i++) {
-                            JSC::JSValue transferListValue = transferListArray->get(lexicalGlobalObject, i);
-                            if (transferListValue.isObject()) {
-                                JSC::JSObject* transferListObject = transferListValue.getObject();
-                                transferList.append(JSC::Strong<JSC::JSObject>(vm, transferListObject));
-                            }
+        if (JSValue transferListValue = optionsObject->getIfPropertyExists(lexicalGlobalObject, Identifier::fromString(vm, "transferList"_s))) {
+            if (transferListValue.isObject()) {
+                JSC::JSObject* transferListObject = transferListValue.getObject();
+                if (auto* transferListArray = jsDynamicCast<JSC::JSArray*>(transferListObject)) {
+                    for (unsigned i = 0; i < transferListArray->length(); i++) {
```

Comment:
I think there's a forEach function wrapper thing we can do here that is slightly faster. specifically, it handles the getIndexQuickly case